From 05259336358790ebe0d1d3822c691f9fe291e536 Mon Sep 17 00:00:00 2001
From: wu guoxing <b39297@freescale.com>
Date: Tue, 27 Dec 2011 13:32:43 +0800
Subject: ENGR00170945:mx6: clk: __clk_disable operates only when usecount > 0

in __clk_disable, check usecount, if it is 0, return, otherwise,
the usecount will be un-correct.

Signed-off-by: Wu Guoxing <b39297@freescale.com>
---
 arch/arm/plat-mxc/clock.c |    6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/arch/arm/plat-mxc/clock.c b/arch/arm/plat-mxc/clock.c
index 1814cdc..e192357 100755
--- a/arch/arm/plat-mxc/clock.c
+++ b/arch/arm/plat-mxc/clock.c
@@ -65,7 +65,11 @@ static void __clk_disable(struct clk *clk)
 {
 	if (clk == NULL || IS_ERR(clk))
 		return;
-	WARN_ON(!clk->usecount);
+
+	if (!clk->usecount) {
+		WARN(1, "clock enable/disable mismatch!\n");
+		return;
+	}
 
 	if (!(--clk->usecount)) {
 		if (clk->disable)
-- 
1.7.9.5

